src/libraries/System.Private.Xml/src/System/Xml/Schema/DataTypeImplementation.cs

```diff
@@ -1126,8 +1126,12 @@ internal override RestrictionFlags ValidRestrictionFlags
                     values.Add(typedValue);
                 }
                 Debug.Assert(_itemType.ListValueType.GetElementType() == _itemType.ValueType);
-                array = ToArray(values, _itemType.ListValueType);
+                arrayType = _itemType.ListValueType;
             }
+
+            Array array = Array.CreateInstanceFromArrayType(arrayType, values.Count);
```

Comment:
Nit: This can be after `if (values.Count < _minListSize)` check. We do not need to create the array when returning an exception.